src/main/2824.c

```diff
@@ -119,7 +127,18 @@ void func_8001293C(void)
 {
 }
 
-INCLUDE_ASM("asm/us/main/nonmatchings/2824", func_80012944);
+void func_80012944(void)
+{
+    s32 i;
+
+    for (i = 0; i < 8; i++) {
+        (i + *(s16**)0x801F8300)[48] = 0;
```

Comment:
I'm not sure what it is yet but this area seems to be related to changing threads https://github.com/sozud/mmx4/blob/79b403c47018ac7de9649fca3e3dbb405294905c/src/main/2824.c#L94